starfish/pipeline/features/pixels/combine_adjacent_features.py

```diff
@@ -0,0 +1,117 @@
+        thought of as the number of square steps allowed between pixels that are considered
+        connected. 1 enforces adjacent pixels. 2 (default) allows diagonal pixels in the same
+        plane. 3 allows diagonal pixels in 3d.
+
+    Returns
+    -------
+    IntensityTable :
+        spot intensities
+    List[Object] :
+        region properties for each spot (see skimage.measure.regionprops)
+    np.ndarray[int] :
+        label image wherein each connected component (spot) is coded with a different
+        integer
+    np.ndarray[int] :
+        decoded image wherein each target is coded as a different integer. Intended for visualization
+
+    """
+    # None needs to map to zero, non-none needs to map to something else.
+    int_to_gene = dict(
+        zip(range(1, np.iinfo(np.int).max),
+            set(intensities.gene_name.values) - {'None'}))
+    int_to_gene[0] = 'None'
+    gene_to_int = {v: k for (k, v) in int_to_gene.items()}
+
+    # map genes to ints
+    gene_list = [gene_to_int[g] for g in intensities.coords[intensities.Constants.GENE.value].values]
+    gene_array = np.array(gene_list)
+
+    # reverses the linearization that was used to construct the IntensityTable from the ImageStack
+    decoded_image: np.ndarray = gene_array.reshape(intensities.attrs['image_shape'])
+
+    # label each pixel according to its component
+    label_image: np.ndarray = label(decoded_image, connectivity=connectivity)
+
+    # calculate the mean value grouped by (ch, round) across all pixels of a connected component
+    pixel_labels = label_image.reshape(-1)
+    intensities['spot_id'] = (IntensityTable.Constants.FEATURES.value, pixel_labels)
+    mean_pixel_traces = intensities.groupby('spot_id').mean('features')
+
+    # "Zero" spot id is background in the label_image. Remove it.
+    mean_pixel_traces = mean_pixel_traces.loc[mean_pixel_traces['spot_id'] > 0]
+    props: List = regionprops(np.squeeze(label_image))
+
+    # calculate spots and drop ones that fail the area threshold
+    spots = []
+    passes_filter = np.ones_like(props, dtype=np.bool)  # default is passing (ones)
+    for i, spot_property in enumerate(props):
+        if spot_property.area >= area_threshold:
+
+            # because of the above skimage issue, we need to support both 2d and 3d properties
+            if len(spot_property.centroid) == 3:
+                spot_attrs = {
+                    'z': int(spot_property.centroid[0]),
+                    'y': int(spot_property.centroid[1]),
+                    'x': int(spot_property.centroid[2])
+                }
+            else:  # data is 2d
+                spot_attrs = {
+                    'z': 0,
+                    'y': int(spot_property.centroid[0]),
+                    'x': int(spot_property.centroid[1])
+                }
+
+            # we're back to 3d or fake-3d here
+            gene_index = decoded_image[spot_attrs['z'], spot_attrs['y'], spot_attrs['x']]
```

Comment:
i want everything un-filtered, with a boolean that tells me what passed the filters. i also want qualities (or distances)

Reply:
#347 for un-filtered, distances have been added back.